### PR TITLE
MULE-15277 - Improve documentation about application and domains for …

### PR DESCRIPTION
**Pull request by pablolagreca**
…Mule 4

### CODE REVIEW

File: mule4-user-guide/v/4.1/intro-packaging.adoc [hunk 1]
```diff
@@ -48,10 +48,30 @@ the required Mule version, and the class loader configuration.
...
+| minMuleVersion | string | the minimum mule runtime version required to deploy the application |
+| requiredProduct | string | the required product type to deploy the application. MULE means that the application can be deployed on the CE version or the EE version, MULE_EE means that the application can only be deployed on the EE version |
+| classLoaderModelLoaderDescriptor | object | descriptor of the classloading model for the application. The id field identifies the mechanism to be used by the runtime to understand how the dependencies and the packages/resources of the application are meant to be used. By default mule uses the id `mule` which means that the exported packages and resources from the applications are described in the `attributes` field and the dependencies are described in the file `/META-INF/mule-artifact/classloader-model.json` |
+| bundleDescriptorLoader | object | descriptor of the application bundle coordiantes. The ide field identifies the mechanism to be used by the runtime to understand the application bundle coordinates. The default one used is `mule` and will load the artifact group id, artifact id and version from the pom file. |
```
Comment (gsfernandes) [inline mule4-user-guide/v/4.1/intro-packaging.adoc]: ide -> id

File: mule4-user-guide/v/4.1/intro-packaging.adoc [hunk 2]
```diff
@@ -48,10 +48,30 @@ the required Mule version, and the class loader configuration.
...
+| secureProperties | array of string | declares the set of configuration properties of the artifact that must be managed as secure in the platform |
+|===
+
+== Application versioning
+
+Mule Runtime follows link:https://semver.org/[semantic versioning] in all it's artifacts. By following semantic versioning clients of our APIs can cleary understand what to expect whenever we release a new version of an asset. For instance, consider the case of mule connectors. You can expect just bug fixes when just the patch version is updated. You can expected new feature in new minor versions. Both of those changes are expected to be backward compatible so you can upgrade without any problem. But if we update the major version of the connector it means that we had to break backward compatibility to include new funcionality or provide a much better UX.
```
Comment (gsfernandes) [inline mule4-user-guide/v/4.1/intro-packaging.adoc]: Don't know if the use of 'we' would be aligned with the rest of the docs. Is someone else from docs going to rephrase this?

File: mule4-user-guide/v/4.1/intro-packaging.adoc (hunk 1 above)
Comment (dukesphere) [reply mule4-user-guide/v/4.1/intro-packaging.adoc]: fixed this.

File: mule4-user-guide/v/4.1/intro-packaging.adoc (hunk 2 above)
Comment (dukesphere) [reply mule4-user-guide/v/4.1/intro-packaging.adoc]: agree about use of "we" but left it in place for now. Should we say "MuleSoft", instead?

**Comment by dukesphere** [comment]
Please see if the copyedits address your concerns @gsfernandes.

**Pull request merged**